app/src/main/java/com/example/tasky/feature_authentication/presentation/register_screen/RegisterScreen.kt

```diff
@@ -0,0 +1,231 @@
+
+    var fullNameError by remember {
+        mutableStateOf("")
+    }
+    var emailError by remember {
+        mutableStateOf("")
+    }
+    var passwordError by remember {
+        mutableStateOf("")
+    }
+
+    LaunchedEffect(key1 = true) {
+        viewModel.authResult.collect { result ->
+            when(result) {
+                is AuthUseCaseResult.GenericError -> {
+                    Toast.makeText(context, result.message, Toast.LENGTH_LONG).show()
+                }
+                is AuthUseCaseResult.Success -> {
+                    navController.navigate(Screen.AgendaScreen.route)
+                }
+                else -> Unit
+            }
+        }
+    }
+
+    LaunchedEffect(key1 = true) {
+        viewModel.fullNameResult.collect {
+            fullNameError = it.getString(context)
+        }
+    }
+
+    LaunchedEffect(key1 = true) {
+        viewModel.emailResult.collect {
+            emailError = it.getString(context)
+        }
+    }
+
+    LaunchedEffect(key1 = true) {
+        viewModel.passwordResult.collect {
+            passwordError = it.getString(context)
+        }
+    }
+
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = Modifier
+            .fillMaxSize()
+            .background(BackgroundBlack)
+    ) {
+        Text(
+            text = stringResource(id = R.string.RegisterScreenTitle),
+            color = Color.White,
+            style = MaterialTheme.typography.titleLarge,
+            modifier = Modifier.padding(top = 77.dp)
+        )
+        Spacer(modifier = Modifier.height(72.dp))
+        Box(modifier = Modifier
+            .fillMaxSize()
+            .clip(
+                RoundedCornerShape(
+                    topStart = 30.dp,
+                    topEnd = 30.dp
+                )
+            )
+            .background(BackgroundWhite)
+        ) {
+            Column(
+                horizontalAlignment = Alignment.CenterHorizontally,
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(
+                        top = 40.dp,
+                        start = 16.dp,
+                        end = 16.dp,
+                    )
+            ) {
+                TaskyTextField(
+                    value = state.fullName,
+                    onValueChanged = {
+                        viewModel.onEvent(RegisterEvent.FullNameChanged(it))
+                    },
+                    hint = stringResource(id = R.string.FullNameHint),
+                    keyboardType = KeyboardType.Text,
+                    contentDescription = stringResource(id = R.string.DescriptionFullNameValid),
+                    isValid = state.isFullNameValid,
+                    isError = state.fullNameError != null
+                )
+                if(state.fullNameError != null) {
+                    Text(text = fullNameError, color = RedInvalid)
+                }
```

Comment:
The Compose `TextField` should already have a property for the error message, but you can also show it like this, that's fine